### DISCUSSION TITLE
Kamera Snapshot in DisplayImage funktioniert nur einmal

### DISCUSSION THREAD

**Discussion by MarkSau** [Zefau/ioBroker.jarvis#1151] [Q&A] [answered]
Hi.
Ich möchte mir ein Bild, erzeugt von Motioneye, in einem Displayimage zeigen lassen. Die Kamera ist eine Reolink. Leider kann ich keiner der möglichen Urls, die mir die Kamera bietet in Jarvis einbinden. Daher der Umweg über Motioneye.

Das Bild wird auch einmal gezeigt. Allerdings nach dem eingestellten Aktualisierungsintervall, 10sek, ist das Bild leer und im Browser kommen folgende Fehlermeldungen

![image](https://user-images.githubusercontent.com/22911084/139928697-148a4da3-41a7-44f9-8df3-7cee1c484e20.png)

Dazu habe ich folgendes erstellt:

![image](https://user-images.githubusercontent.com/22911084/139928932-d748f550-4bb0-46d4-9aa9-1a92d68a5b79.png)

Aber vielleicht ist es ja auch der falsche Weg. 

**Zusatz**: mit dem iframe widget wird das Bild zu gross angezeigt (scrollbalken). Ich habe hier aber keine Idee, wie ich das resizen kann.

**Comment by Zefau** [comment]
Der Zusatz `?r=1..` kommt von jarvis, um eine Aktualisierung zu erzwingen. Die URL in dem Fehler scheint aber auch keine Grafik-Datei zu sein. Die URL sieht aus wie ein Ordner? Was kommt denn, wenn du die URL direkt im Browser öffnest? Was kommt, wenn du ohne `?r=1..`  im Browser öffnest?

**Reply by MarkSau** [reply]
Wenn ich die Url ohne ?r=1.. eingebe, kommt aus meiner Sicht natürlich, auch ein 403...Weil die Motioneye ja den Zusatz /?_username=.... verlangt. Das Problem wird sein, das ich ja kein Bild direkt, sondern Motioneye nach dem aktuellen Bild frage, was dann ja erst erzeugt und angezeigt wird. 
Wie ich oben beschrieben habe, kann ich die Url auch bei iframe benutzen. Da habe ich aber eben das Prob, das ich keine Möglichkeit habe die Größe des Bilds anzupassen. Weder bei der Abfrage noch im Widget selbst. Oder auch weiß ich nicht wie?

**Comment by MarkSau** [comment]
Ich habe mal die direkte URL der Kamera ins Display Image eingefügt. Auch hier erzeugt der Refresh einen Fehler, allerdings einen anderen:
Original Link:  http://192.168.1.91/cgi-bin/api.cgi?cmd=Snap&channel=0&rs=wuuPhkmUCeI9WG7C&user=admin&password=XXXXXXX&width=640&height=480

Fehler nach 20 Sekunden

![image](https://user-images.githubusercontent.com/22911084/140290196-dc0d50bd-4a71-44c0-8269-c0c6df7ede98.png)

Ich denke es würde helfen, wenn jarvis den original link beim refresh nutzen würde.

Zusatz:  der Link direkt http://192.168.1.91/cgi-bin/api.cgi?r=1636018380526 zeigt:

[ { "cmd" : "Unknown", "code" : 1, "error" : { "detail" : "please login first", "rspCode" : -6 } } ]

**Reply by Zefau** [reply]
Ich kann den originalen Link nicht nutzen, weil das keinen refresh erwingt (https://github.com/Zefau/ioBroker.jarvis/issues/1146)

**Comment by typedefChris** [comment, marked as answer]
Mahlzeit,
du musst mit der Reolink nicht den Umweg über MotionEye gehen.
Es ist möglich in dem Link die Parameter zu vertauschen.
Ursprüngliche Vorgabe von Reolink:
_http://(ip address)/cgi-bin/api.cgi?cmd=Snap&channel=0&rs=(any combination of
numbers and letters)&user=(user name)&password=(user password)_
Diese kann aber beliebig geändert werden, sodass der rs-code, der zwischen jeder Aktualisierung verändert werden muss, am Ende steht:
_http://(ip address)/cgi-bin/api.cgi?cmd=Snap&channel=0&user=(user name)&password=(user password)&rs=(any combination of
numbers and letters)_
Dann nur noch im DisplayImage entsprechend anpassen und es sollte laufen:
![grafik](https://user-images.githubusercontent.com/106241676/170245996-b90ba24a-ea45-4996-b304-76a17656ee13.png)

*_Any combination of numbers and letters_ ist wörtlich zu nehmen. Hier kann eine beliebige Kombination eingetragen werden

**Reply by MarkSau** [reply]
Das funktioniert....Danke für den super Hinweis...
